### PR TITLE
(chore) O3-3574 : Refactor E2E test for appointments

### PR DESCRIPTION
**Pull request by chimanwadike**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR refactors existing e2e tests for appointments by writing separate tests for adding, updating, and canceling.
This is to increase test clarity and visibility. 

## Related Issue

https://issues.openmrs.org/browse/O3-3574

### CODE REVIEW

File: e2e/specs/appointments.spec.ts [hunk 1]
```diff
@@ -7,61 +7,138 @@ import { type Visit } from '@openmrs/esm-framework';
...
-test.beforeEach(async ({ api }) => {
+test.beforeAll(async ({ api }) => {
   patient = await generateRandomPatient(api);
   visit = await startVisit(api, patient.uuid);
 });
 
-test('Add an appointment', async ({ page, api }) => {
-  const appointmentsPage = new AppointmentsPage(page);
+test.describe.serial('Running appointment tests sequentially', () => {
+  test('Add an appointment', async ({ page }) => {
```
Comment (chimanwadike) [inline e2e/specs/appointments.spec.ts:16]: @jayasanka-sack @kdaud I would like to seek your views concerning the use of `test.describe` in scenario of sequential operations e.g. `create, edit and delete`  like this one.
This is to improve some of our e2e tests going forward and prepare towards writing more advanced test cases.

Comment (kdaud) [reply e2e/specs/appointments.spec.ts:16]: @chimanwadike the refactor should involve decoupling the test scenarios for adding, editing, and canceling an appointment.

**Review by kdaud — COMMENTED** [review]
@chimanwadike the refactor should involve decoupling the test scenarios for adding, editing, and canceling an appointment.

**Comment by chimanwadike** [comment]
> @chimanwadike the refactor should involve decoupling the test scenarios for adding, editing, and canceling an appointment.

@kdaud this is noted.

**Comment by chimanwadike** [comment]
> @chimanwadike the refactor should involve decoupling the test scenarios for adding, editing, and canceling an appointment.

@kdaud I've been away for a while now. Following your last comment, it appears we would have to create separate appointments to test edit and cancel if these test cases must be decoupled that far. 

In order to decouple the tests the way you've explained and how I understand it, we could run into overheads of more API calls and could lead to slower tests in the grand scheme which is neither what we need.

I'm thinking there maybe no need for this PR, what do you think?

**Comment by brandones** [comment]
@kdaud What do you think of what @chimanwadike wrote? Should we close this PR?

**Comment by brandones** [comment]
Ok, @kdaud ?

**Comment by kdaud** [comment]
>  it appears we would have to create separate appointments to test edit and cancel if these test cases must be decoupled that far.

Yes, this is possible via the API. See an example [here](https://github.com/openmrs/openmrs-esm-patient-management/blob/main/e2e/commands/visit-operations.ts#L5-L18)

**Comment by brandones** [comment]
Ok @chimanwadike , is it clear to you how to proceed?

**Comment by chimanwadike** [comment]
> Ok @chimanwadike , is it clear to you how to proceed?

Yes it is.
thanks

**Pull request closed without merging**